app/controllers/comments_controller.rb

```diff
@@ -1,4 +1,5 @@
 class CommentsController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
- [ ] **[OPTIONAL]** I can see you are adding this method to every controller. But if you observe closely you will notice that all these controllers inherit from the `ApplicationController`. So instead of repeating the method you can just add this method in the application controller and remove this method everywhere else in the controllers. The authentication would work as you expect.

Reply:
Thanks